### PR TITLE
new: api server specific fields in KuberneteCluster

### PR DESCRIPTION
**Pull request by sibicramesh**
Adding new fields specific to k8s api server

### CODE REVIEW

File: specs/kubernetescluster.spec [hunk 1]
```diff
@@ -29,19 +29,36 @@ model:
 # Attributes
 attributes:
   v1:
-  - name: APIVersions
-    description: API versions supported by the API server.
+  - name: APIServerServiceFQDNs
+    description: It contains the fqdns that will be set in the certificate SANS field.
```
Comment (primalmotion) [inline specs/kubernetescluster.spec:33]: `Contains the FQDNs used by the API server. They will be used to populate the Certificate DNS SANs field`

File: specs/kubernetescluster.spec [hunk 2]
```diff
@@ -29,19 +29,36 @@ model:
...
+    type: list
+    exposed: true
+    subtype: string
+    stored: true
+    orderable: true
+
+  - name: APIServerServiceIPs
+    description: It contains the ips that will be set in the certificate SANS field.
```
Comment (primalmotion) [inline specs/kubernetescluster.spec:41]: Contains the IPs used by the API server. They will be used to populate the Certificate IP SANs field

File: specs/kubernetescluster.spec [hunk 3]
```diff
@@ -29,19 +29,36 @@ model:
...
     type: list
     exposed: true
     subtype: string
     stored: true
 
-  - name: K8SNamespace
-    description: Kubernetes namespace.
+  - name: APIServerServiceName
+    description: |-
```
Comment (primalmotion) [inline specs/kubernetescluster.spec:48]: if we want to impose a format, we must add a custom validation

File: specs/kubernetescluster.spec [hunk 4]
```diff
@@ -29,19 +29,36 @@ model:
...
+      Kubernetes service name in the format <service name>.<service name
+      namespace>.svc will be set in the certificate CommonName field.
     type: string
     exposed: true
     stored: true
 
+  - name: APIServerVersions
```
Comment (primalmotion) [inline specs/kubernetescluster.spec:55]: This probably needs a validation? I'm nor sure how an APIServer version looks like

**Review by primalmotion — CHANGES_REQUESTED** [review]

File: specs/kubernetescluster.spec (hunk 4 above)
Comment (sibicramesh) [reply specs/kubernetescluster.spec:61]: The version can have different schemas, there is no definite format for it. Here is the doc https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#specify-multiple-versions

**Review by philipatl — APPROVED** [review]

File: custom_validations.go [hunk 5]
```diff
@@ -1475,6 +1475,29 @@ func ValidatePortsList(attribute string, ports []*portutils.PortsRange) error {
 	return nil
 }
 
+// ValidateAPIServerServiceName validates the api server service name.
+func ValidateAPIServerServiceName(attribute string, serviceName string) error {
+
+	parts := strings.Split(serviceName, ".")
+	if len(parts) != 3 {
+		return makeValidationError(attribute, fmt.Sprintf("invalid service name, should be of format <service name>.<service snamespace>.svc: %s", serviceName))
```
Comment (primalmotion) [inline custom_validations.go:1483]: snamespace

**Review by primalmotion — CHANGES_REQUESTED** [review]
just a minor typo. otherwise LGTM

**Review by philipatl — DISMISSED** [review]

**Review by philipatl — APPROVED** [review]

**Pull request merged**